### PR TITLE
Add note about cert-manager's import path

### PR DESCRIPTION
**Pull request by SgtCoDFish**
Part of [the repo migration followups](https://github.com/cert-manager/cert-manager/issues/4824).

### CODE REVIEW

**Comment by SgtCoDFish** [comment]
See https://deploy-preview-833--cert-manager-website.netlify.app/docs/contributing/importing/

File: content/en/docs/contributing/importing.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+title: "Importing cert-manager in Go"
+linkTitle: "Importing cert-manager in Go"
+weight: 500
+type: "docs"
+---
+
+cert-manager is written in Go, and uses Go modules. You _can_ import it as a go module, and in some cases
```
Comment (jsoref) [inline content/en/docs/contributing/importing.md:8]:
```suggestion
cert-manager is written in Go, and uses Go modules. You _can_ import it as a Go module, and in some cases
```

File: content/en/docs/contributing/importing.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+If you think you really need to import other parts of the code, please do reach out and talk to us so we're aware of this
+need! We'll always try to avoid breakage where we can.
+
+## Module Import Paths
+
+The original cert-manager repository was created on GitHub as `https://github.com/jetstack/cert-manager`, and after several
+years was migrated to `https://github.com/cert-manager/cert-manager`.
+
+This means the go module import path you need depends on the version of cert-manager you're trying to use.
```
Comment (jsoref) [inline content/en/docs/contributing/importing.md:38]:
```suggestion
This means the Go module import path you need depends on the version of cert-manager you're trying to use.
```

File: content/en/docs/contributing/importing.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+## When You Might Import cert-manager
+
+You might need to import cert-manager if you're writing Go code which:
+
+- uses cert-manager custom resources, so you want to import something under `pkg/apis`
+- implements an external DNS solver webhook, as in the [webhook-example](https://github.com/cert-manager/webhook-example)
+- implements an external issuer, as in the [sample-external-issuer](https://github.com/cert-manager/sample-external-issuer)
+
+If you think you really need to import other parts of the code, please do reach out and talk to us so we're aware of this
```
[the remark this answers is not in the document]
Comment (SgtCoDFish) [reply content/en/docs/contributing/importing.md]: Good idea, thanks!

**Comment by SgtCoDFish** [comment]
I think I've addressed the comments now @jsoref :grin:

File: content/en/docs/contributing/importing.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+- implements an external DNS solver webhook, as in the [webhook-example](https://github.com/cert-manager/webhook-example)
+- implements an external issuer, as in the [sample-external-issuer](https://github.com/cert-manager/sample-external-issuer)
+
+If you think you really need to import other parts of the code, please do reach out and [talk to us](../#slack) so we're
+aware of this need! We'll always try to avoid breakage where we can.
+
+## Module Import Paths
+
+The original cert-manager repository was created on GitHub as `https://github.com/jetstack/cert-manager`, and after several
+years was migrated to `https://github.com/cert-manager/cert-manager`.
```
Comment (jsoref) [inline content/en/docs/contributing/importing.md:36]: I'm not sure the value of `several years` vs. just giving a creation date and a migration date - or just saying `later` or mentioning that it was migrated before version 1.8.

Comment (SgtCoDFish) [reply content/en/docs/contributing/importing.md:36]: Yeah that's a fair point, there's not really any value provided by "several years" here. I'll remove!

**Review by erikgb — APPROVED** [review]
LGTM

**Review by wallrj — APPROVED** [review]
Thanks @SgtCoDFish 

Nice explanation.

The [cert-manager Readme](https://github.com/cert-manager/cert-manager#importing-cert-manager-as-a-module) hints that cert-manager APIs follow the [Kubernetes API Deprecation Policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/).
We could add a note about that here or in a followup PR.

/lgtm
/hold in case you want to make any further changes.

**Comment by SgtCoDFish** [comment]
/unhold

Thanks all!

**Pull request merged**